### PR TITLE
adding stack cookie support

### PR DESCRIPTION
**Pull request by Flickdm**
# Preface

Please ensure you have read the [contribution docs](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md) prior
to submitting the pull request. In particular,
[pull request guidelines](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md#pull-request-best-practices).

## Description

[Issue 118](https://github.com/microsoft/mu_oem_sample/issues/118)

For each item, place an "x" in between `[` and `]` if true. Example: `[x]`.
_(you can also check items in the GitHub UI)_

- [X] Impacts functionality?
  - Yes, builds with stack cookies by default for OemPkg when using `stuart_ci_build`
- [X] Impacts security?
  - Yes, adds stack cookies
- [ ] Breaking change?
  - This shouldn't be breaking
- [ ] Includes tests?
  - No
- [ ] Includes documentation?
  - No

## How This Was Tested

This was built in a windows environment with VS2022 using stuart_ci_build and built on a linux environment using WSL2 on ubuntu 22.04.2 using GCC5

## Integration Instructions

N/A

### CODE REVIEW

File: OemPkg/OemPkg.dsc [hunk 1]
```diff
@@ -85,6 +85,9 @@
   ConfigVariableListLib|SetupDataPkg/Library/ConfigVariableListLib/ConfigVariableListLib.inf
   ActiveProfileIndexSelectorLib|OemPkg/Library/ActiveProfileIndexSelectorPcdLib/ActiveProfileIndexSelectorPcdLib.inf
 
+  # Add support for GCC stack protector
+  NULL|MdePkg/Library/BaseStackCheckLib/BaseStackCheckLib.inf
```
Comment (kuqin12) [inline OemPkg/OemPkg.dsc]: I think we should put this under `[LibraryClasses.AARCH64]`

**Review by kuqin12 — APPROVED** [review]

**Review by makubacki — APPROVED** [review]

File: OemPkg/OemPkg.dsc (hunk 1 above)
Comment (os-d) [reply OemPkg/OemPkg.dsc]: Is it not valid for x86?

**Review by os-d — APPROVED** [review]

File: OemPkg/OemPkg.dsc (hunk 1 above)
Comment (kuqin12) [reply OemPkg/OemPkg.dsc]: This stack cookie only has implementation for GCC (I think even that implementation is a pseudo one without legit random number support).

The x64 stack cookie related symbols for VS builds are injected through `MdePkg/Library/BaseBinSecurityLibRng/BaseBinSecurityLibRng.inf`.

Comment (os-d) [reply OemPkg/OemPkg.dsc]: Right, I'm thinking that x64 builds on GCC may be broken?

Comment (kuqin12) [reply OemPkg/OemPkg.dsc]: Ah, yeah, that is set here: https://github.com/microsoft/mu_basecore/blob/347cce4949a5073b284d7d73d307797659033ff0/BaseTools/Conf/tools_def.template#L2075, where x86 builds disabled stack guards with GCC.

**Pull request merged**